cleverhans/utils.py

```diff
@@ -220,6 +221,92 @@ def grid_visual(data):
     return figure
 
 
+def get_logits_over_interval(sess, model, x, adv_x_real, x_data,
+                             min_epsilon=-10, max_epsilon=10,
+                             num_points=21):
+    """Get logits when the input is perturbed in an interval in adv direction.
+
+    Args:
+        sess: Tf session
+        model: Model for which we wish to get logits
+        x: Tf placeholder for data (unquantized)
+        adv_x_real: Tf placeholder for adv perturbation of x (unquantized)
+        x_data: Numpy array corresponding to the data slice used to generate plot
+        min_epsilon: Minimum value of epsilon over the interval
+        max_epsilon: Maximum value of epsilon over the interval
+        num_points: Number of points used to interpolate
+
+    Returns:
+        Numpy array containing log probabilities
+    """
+    delta = (max_epsilon - min_epsilon) / (num_points - 1)
+    epsilon = min_epsilon
+    eta = adv_x_real - x
+    eta_normalized = tf.nn.l2_normalize(eta, dim=0)
+
+    epsilon_t = tf.placeholder(tf.float32, shape=(), name="epsilon")
+    adv_x = x + epsilon_t * eta_normalized
+    logits = model.get_logits(adv_x)
+    for i in xrange(num_points):
```

Comment:
Rather than using a `for` loop on xrange(num_points), would it make sense to use `np.arange` to improve readability?